python/mxnet/ndarray/ndarray.py

```diff
@@ -2925,6 +2925,49 @@ def imdecode(str_img, clip_rect=(0, 0, 0, 0), out=None, index=0, channels=3, mea
                                    out=out)
 
 
+def eye(N, M=0, k=0, ctx=None, dtype=None, **kwargs):
+    """Return Return a 2-D array with ones on the diagonal and zeros elsewhere.
+
+    Parameters
+    ----------
+    N: int
+        Number of rows in the output.
+    M: int, optional
+        Number of columns in the output. If 0, defaults to N.
+    k: int, optional
+        Index of the diagonal: 0 (the default) refers to the main diagonal,
+        a positive value refers to an upper diagonal,
+        and a negative value to a lower diagonal.
+    ctx: Context, optional
+        An optional device context (default is the current default context)
+    dtype: str or numpy.dtype, optional
+        An optional value type (default is `float32`)
+    aux_types: list of numpy.dtype, optional
+        An optional list of types of the aux data for RowSparseNDArray or CSRNDArray
+        (default values depends on the storage type)
+
+    Returns
+    -------
+        A created array
+
+    Examples
+    --------
+    >>> mx.nd.eye(1, 2)
+
+    [[ 1.  0.]]
+    <NDArray 1x2 @cpu(0)>
+    >>> mx.nd.eye(1, 2, 1).asnumpy()
+    array([[ 0.,  1.]], dtype=float32)
+    """
+    # pylint: disable= unused-argument
+    if ctx is None:
+        ctx = Context.default_ctx
+    dtype = mx_real_t if dtype is None else dtype
+    # pylint: disable= no-member, protected-access
```

Comment:
are these pylint disable necessary?

Reply:
I just follow the way in `zeros`. https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/ndarray/ndarray.py#L2928